README.md

```diff
@@ -1,23 +1,151 @@
+| `zsh`        | Generate script for Zsh                     |
+
+#### `generate`
+
+Commands related to Kubernetes Gateway API and Kuadrant resource generation from OpenAPI 3.x specifications.
+
+| Subcommand   | Description                                   |
+| ------------ | --------------------------------------------- |
+| `gatewayapi` | Generate Gateway API resources                |
+| `kuadrant`   | Generate Kuadrant resources                   |
+
+##### `generate gatewayapi`
+
+Generate Gateway API resources from an OpenAPI 3.x specification
+
+| Subcommand | Description                                      | Flags                             |
+| ---------- | ------------------------------------------------ | --------------------------------- |
+| `httproute`| Generate Gateway API HTTPRoute from OpenAPI 3.0.X| `--oas string` Path or URL to OpenAPI spec (required) |
+
+##### `generate kuadrant`
+
+Generate Kuadrant resources from an OpenAPI 3.x specification
+
+| Subcommand       | Description                                       | Flags                             |
+| ---------------- | ------------------------------------------------- | --------------------------------- |
+| `authpolicy`     | Generate a [Kuadrant AuthPolicy](https://docs.kuadrant.io/kuadrant-operator/doc/auth/) from an OpenAPI 3.0.x specification   | `--oas string` Path or URL to OpenAPI spec (required) |
+| `ratelimitpolicy`| Generate [Kuadrant RateLimitPolicy](https://docs.kuadrant.io/kuadrant-operator/doc/rate-limiting/) from an OpenAPI 3.0.x specification | `--oas string` Path or URL to OpenAPI spec (required) |
+
+
+#### `install`
+
+Install the [Kuadrant Operator](https://github.com/Kuadrant/kuadrant-operator) into an OLM-powered cluster.
+
+| Flag                  | Description                      |
+| --------------------- | -------------------------------- |
+| `--kubeconfig` string | Kubernetes configuration file    |
+
+* For more information of the Kuadrant Operator, see https://docs.kuadrant.io/kuadrant-operator/
+
+#### `uninstall`
+
+Remove Kuadrant from the cluster.
+
+| Flag                  | Description                      |
+| --------------------- | -------------------------------- |
+| `--kubeconfig` string | Kubernetes configuration file    |
+
+#### `version`
+
+Print the version number of `kuadrantctl`.
+
+No additional flags or subcommands.
+
+### Additional Guides
+
+#### Generating Gateway API HTTPRoute Objects
+
+- Generates [Gateway API HTTPRoute](https://gateway-api.sigs.k8s.io/v1alpha2/guides/http-routing/) objects from an OpenAPI Specification (OAS) 3.x.
+- Supports reading from a file, URL, or stdin.
+- Example usages and more information can be found in the [detailed guide](doc/generate-gateway-api-httproute.md).
+
+#### Generating Kuadrant AuthPolicy Objects
+
+- Generates [Kuadrant AuthPolicy](https://github.com/Kuadrant/kuadrant-operator/blob/v0.4.1/doc/auth.md) objects for managing API authentication.
```

Comment:
why version 0.4.1 for the absolute links?

Reply:
Thought I'd leave these links at 0.4.1 right now as that's what `kuadrantctl install` installs right now (until https://github.com/Kuadrant/kuadrantctl/issues/56 is fixed)

Reply:
ah ok makes sense maybe a note so we don't forget?